File: src/main/java/com/frank/entity/book/BookAvailability.java
package com.frank.entity.book;

class BookAvailability {

    private int availability;

    private BookAvailability(int availability) {
        this.availability = availability;
    }

    public int getAvailability() {
        return availability;
    }

    public void increase() {
        this.availability += 1;
    }

    final public static BookAvailability ZERO = new BookAvailability(0);
    final public static BookAvailability ONE = new BookAvailability(1);

}

File: src/main/kotlin/com/frank/entity/book/BookAvailability.kt
package com.frank.entity.book

internal class BookAvailability private constructor(availability: Int) {

    var availability: Int = 0
        private set

    init {
        this.availability = availability
    }

    fun increase() {
        this.availability += 1
    }

    companion object {

        val ZERO = BookAvailability(0)
        val ONE = BookAvailability(1)
    }

}
